extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -755,20 +780,35 @@ export class AmpConsent extends AMP.BaseElement {
   }
 
   /**
-   * TODO (micajuineho): Use our stored info to check if we have the
-   * necessary granular consents.
-   * @param {ConsentInfoDef} unusedConsentInfo
+   * Returns true if we have stored the granular consent values
+   * for the required purposes.
+   * @param {ConsentInfoDef} consentInfo
    * @return {!Promise<boolean>}
    */
-  checkGranularConsentRequired_(unusedConsentInfo) {
+  checkGranularConsentRequired_(consentInfo) {
     if (!this.isGranularConsentExperimentOn_) {
       return Promise.resolve(true);
     }
     return this.getPurposeConsentRequired_().then((purposeConsentRequired) => {
-      if (!purposeConsentRequired) {
+      // True if there are no required purposes
+      if (!purposeConsentRequired || !purposeConsentRequired.length) {
         return true;
       }
-      // TODO: add check here.
+      const storedPurposeConsents = consentInfo['purposeConsents'];
+      // False if there are no stored purposes
+      if (
+        !storedPurposeConsents ||
+        !Object.keys(storedPurposeConsents).length
```

Comment:
Driving by, but if you're comparing length you can optimize this  further.

```suggestion
        Object.keys(storedPurposeConsents).length < purposeConsentRequired.length
```